### ISSUE TITLE
push: s3 versioned remote not saving version id for directories

### ISSUE THREAD

**Issue by dberenbaum** [iterative/dvc#8326]
# Bug Report

## Description

After https://github.com/iterative/dvc/pull/8273, `dvc push` adds version IDs for Azure directories but not S3.

### Reproduce

```bash
$ dvc remote add -d cloud s3://dave-sandbox-versioning/remote
Setting 'cloud' as a default remote.
$ dvc remote modify cloud version_aware true
$ dvc remote modify cloud worktree true
$ mkdir dir
$ echo foo > dir/foo.txt
$ echo bar > dir/bar.txt
$ dvc add dir
100% Adding...|█████████████████████████████████████████████████|1/1 [00:00, 21.03file/s]

To track the changes with git, run:

        git add dir.dvc .gitignore

To enable auto staging, run:

        dvc config core.autostage true
$ dvc push
3 files pushed
$ cat dir.dvc
outs:
- md5: cfd746e5595928644e5ff1dc2383674a.dir
  size: 0
  nfiles: 2
  path: dir
  files:
  - size: 4
    etag: c157a79031e1c40f85931829bc5fc552
    md5: c157a79031e1c40f85931829bc5fc552
    relpath: bar.txt
  - size: 4
    etag: d3b07384d113edec49eaa6238ad5ff00
    md5: d3b07384d113edec49eaa6238ad5ff00
    relpath: foo.txt
```

### Expected

A `version_id` field under `files` in `dir.dvc`.

### Environment information

**Output of `dvc doctor`:**

```console
$ dvc doctor
DVC version: 2.26.3.dev13+gd2a31eb66
---------------------------------
Platform: Python 3.8.5 on macOS-10.16-x86_64-i386-64bit
Subprojects:
        dvc_data = 0.10.0
        dvc_objects = 0.4.0
        dvc_render = 0.0.11
        dvc_task = 0.1.2
        dvclive = 0.10.0
        scmrepo = 0.1.1
Supports:
        azure (adlfs = 2022.9.1, knack = 0.7.2, azure-identity = 1.5.0),
        gdrive (pydrive2 = 1.14.0),
        gs (gcsfs = 2022.7.1),
        hdfs (fsspec = 2022.7.1, pyarrow = 3.0.0),
        http (aiohttp = 3.7.3, aiohttp-retry = 2.8.3),
        https (aiohttp = 3.7.3, aiohttp-retry = 2.8.3),
        oss (ossfs = 2021.8.0),
        s3 (s3fs = 2022.7.1, boto3 = 1.21.21),
        ssh (sshfs = 2022.6.0),
        webdav (webdav4 = 0.9.4),
        webdavs (webdav4 = 0.9.4),
        webhdfs (fsspec = 2022.7.1)
Cache types: reflink, hardlink, symlink
Cache directory: apfs on /dev/disk3s1s1
Caches: local
Remotes: local, s3
Workspace directory: apfs on /dev/disk3s1s1
Repo: dvc, git
```

**Comment by pmrowla** [comment]
Issue is because of s3fs dir cache not being invalidated after we push/write files. When we build/collect to get updated version infos after the push, s3fs returns cached info for each pushed file that still has a `None` version ID.

**Comment by pmrowla** [comment]
Secondary problem will be that `info` kwargs are implementation specific in fsspec. `refresh` is used in both s3fs and adlfs to force a non-dircached info call, but is not implemented in all filesystems (like gcsfs). Forcing `refresh` in all filesystems will also negatively impact performance in filesystems that handle version IDs in dircache (adlfs).

**Comment by pmrowla** [comment]
fixed in dvc-data 0.10.1

**Comment by pmrowla** [comment]
related: https://github.com/fsspec/s3fs/issues/653
